contracts/flow/teleport/TeleportCustodyEthereum.cdc

```diff
@@ -1,176 +1,224 @@
-
       self.feeCollector.deposit(from: <-fee)
-
       let amount = vault.balance
       TeleportCustodyEthereum.lockVault.deposit(from: <-vault)
-
       emit Locked(amount: amount, to: to)
       emit FeeCollected(amount: self.lockFee, type: 0)
     }
-
-    pub fun unlock(amount: UFix64, from: [UInt8], txHash: String): @FungibleToken.Vault {
-      pre {
-        !TeleportCustodyEthereum.isFrozen: "Teleport service is frozen"
-        amount <= self.allowedAmount: "Amount unlocked must be less than the allowed amount"
-        amount > self.unlockFee: "Amount unlocked must be greater than unlock fee"
-        from.length == TeleportCustodyEthereum.teleportAddressLength: "Teleport address should be teleportAddressLength bytes"
-        txHash.length == TeleportCustodyEthereum.teleportTxHashLength: "Teleport tx hash should be teleportTxHashLength bytes"
-        !(TeleportCustodyEthereum.unlocked[txHash] ?? false): "Same unlock txHash has been executed"
+    
+    access(AdminEntitlement)
+    fun unlock(amount: UFix64, from: [UInt8], txHash: String): @{FungibleToken.Vault} { 
+      pre { 
+        !TeleportCustodyEthereum.isFrozen:
+          "Teleport service is frozen"
+        amount <= self.allowedAmount:
+          "Amount unlocked must be less than the allowed amount"
+        amount > self.unlockFee:
+          "Amount unlocked must be greater than unlock fee"
+        from.length == TeleportCustodyEthereum.teleportAddressLength:
+          "Teleport address should be teleportAddressLength bytes"
+        txHash.length == TeleportCustodyEthereum.teleportTxHashLength:
+          "Teleport tx hash should be teleportTxHashLength bytes"
+        !(TeleportCustodyEthereum.unlocked[txHash] ?? false):
+          "Same unlock txHash has been executed"
       }
       self.allowedAmount = self.allowedAmount - amount
-
       TeleportCustodyEthereum.unlocked[txHash] = true
       emit Unlocked(amount: amount, from: from, txHash: txHash)
-
       let vault <- TeleportCustodyEthereum.lockVault.withdraw(amount: amount)
       let fee <- vault.withdraw(amount: self.unlockFee)
-
       self.feeCollector.deposit(from: <-fee)
       emit FeeCollected(amount: self.unlockFee, type: 1)
-
-      return <- vault
+      return <-vault
     }
-
-    pub fun withdrawFee(amount: UFix64): @FungibleToken.Vault {
-      return <- self.feeCollector.withdraw(amount: amount)
+    
+    access(AdminEntitlement)
+    fun withdrawFee(amount: UFix64): @{FungibleToken.Vault} { 
+      return <-self.feeCollector.withdraw(amount: amount)
     }
-
-    pub fun updateLockFee(fee: UFix64) {
+    
+    access(AdminEntitlement)
+    fun updateLockFee(fee: UFix64) { 
       self.lockFee = fee
     }
-
-    pub fun updateUnlockFee(fee: UFix64) {
+    
+    access(AdminEntitlement)
+    fun updateUnlockFee(fee: UFix64) { 
       self.unlockFee = fee
     }
-
-    pub fun getFeeAmount(): UFix64 {
+    
+    access(AdminEntitlement)
+    fun getFeeAmount(): UFix64{  
```

Comment:
nit: Will it effect the backend to get the amount? 
(more restrict is better, so this is a nit)

Reply:
fixed.